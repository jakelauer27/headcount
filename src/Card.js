import React from 'react';
import PropTypes from 'prop-types';
import {uid} from 'react-uid';

export default function Card({location, stats, selected, id, selectCard}) {
  const statsKeys = Object.keys(stats);
  let previousStat = 0;
  let headerSize = 'normal-header';

  if (location.length > 15) {
    headerSize = 'small-header'
  }

  const cardStats = statsKeys.map( year => {
    let color = 'green';
    let arrowColor = 'red';
    let direction = 'fa-arrow-down';
    let noChange = !previousStat || stats[year] === previousStat;

    if (stats[year] <= .5) {
      color='red';
    }

    if (stats[year] > previousStat) {
      direction = 'fa-arrow-up';
      arrowColor = 'green'
    }

    if (noChange) {
      arrowColor = 'grey';
      direction = 'fa-minus'
    }

    previousStat = stats[year];

    return (
      <tbody key={uid(year)}>
        <tr className='stat-line'>
          <td className='year'>{year}</td>
          <td className={`value ${color}`}>{(stats[year] * 100).toFixed(0)}%</td>
          <td className='arrow'>
            <i className={`fas ${direction} ${arrowColor}`}></i>
          </td>
        </tr>
      </tbody>
    )
  })

  return (
    <article className={`district-card ${selected}`} onClick={() => selectCard(location)}>
      <div className='left-triangle'></div>
      <div className='right-triangle'></div>
      <div className='content-container'>
        <h1 className={`card-header ${headerSize}`}>{location}</h1>
        <table className='stats-table'>
        {cardStats} 
        </table>
      </div>
    </article> 
  )
} 

Card.propTypes = {
  location: PropTypes.string.isRequired,
  stats: PropTypes.objectOf(PropTypes.number).isRequired,
  id: PropTypes.string.isRequired,
}