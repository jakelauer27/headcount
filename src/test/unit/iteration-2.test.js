import React from 'react'; 
import { shallow } from 'enzyme';
import Card from '../../Card';
import {uid} from 'react-uid';


describe('Card', () => { 
  it('should match the snapshot with all data passed in correctly', () => { 
    const location = 'ASD 20';
    const stats = {'2004': 0.12, '2005': 0.23}

    const wrapper = shallow( <Card 
      location={location}
      stats={stats}
      id={uid(1)}
      key={uid(2)}
    />);

    expect(wrapper).toMatchSnapshot(); 
  }); 
}); 