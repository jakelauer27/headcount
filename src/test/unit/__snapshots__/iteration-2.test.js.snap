// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match the snapshot with all data passed in correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    id="val-1"
    location="ASD 20"
    stats={
        Object {
            "2004": 0.12,
            "2005": 0.23,
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="left-triangle"
/>,
        <div
          className="right-triangle"
/>,
        <div
          className="content-container"
>
          <h1
                    className="card-header normal-header"
          >
                    ASD 20
          </h1>
          <table
                    className="stats-table"
          >
                    <tbody>
                              <tr
                                        className="stat-line"
                              >
                                        <td
                                                  className="year"
                                        >
                                                  2004
                                        </td>
                                        <td
                                                  className="value red"
                                        >
                                                  12
                                                  %
                                        </td>
                                        <td
                                                  className="arrow"
                                        >
                                                  <i
                                                            className="fas fa-minus grey"
                                                  />
                                        </td>
                              </tr>
                    </tbody>
                    <tbody>
                              <tr
                                        className="stat-line"
                              >
                                        <td
                                                  className="year"
                                        >
                                                  2005
                                        </td>
                                        <td
                                                  className="value red"
                                        >
                                                  23
                                                  %
                                        </td>
                                        <td
                                                  className="arrow"
                                        >
                                                  <i
                                                            className="fas fa-arrow-up green"
                                                  />
                                        </td>
                              </tr>
                    </tbody>
          </table>
</div>,
      ],
      "className": "district-card",
      "id": "val-1",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "left-triangle",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "right-triangle",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1
              className="card-header normal-header"
>
              ASD 20
</h1>,
            <table
              className="stats-table"
>
              <tbody>
                            <tr
                                          className="stat-line"
                            >
                                          <td
                                                        className="year"
                                          >
                                                        2004
                                          </td>
                                          <td
                                                        className="value red"
                                          >
                                                        12
                                                        %
                                          </td>
                                          <td
                                                        className="arrow"
                                          >
                                                        <i
                                                                      className="fas fa-minus grey"
                                                        />
                                          </td>
                            </tr>
              </tbody>
              <tbody>
                            <tr
                                          className="stat-line"
                            >
                                          <td
                                                        className="year"
                                          >
                                                        2005
                                          </td>
                                          <td
                                                        className="value red"
                                          >
                                                        23
                                                        %
                                          </td>
                                          <td
                                                        className="arrow"
                                          >
                                                        <i
                                                                      className="fas fa-arrow-up green"
                                                        />
                                          </td>
                            </tr>
              </tbody>
</table>,
          ],
          "className": "content-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "ASD 20",
              "className": "card-header normal-header",
            },
            "ref": null,
            "rendered": "ASD 20",
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <tbody>
                  <tr
                                    className="stat-line"
                  >
                                    <td
                                                      className="year"
                                    >
                                                      2004
                                    </td>
                                    <td
                                                      className="value red"
                                    >
                                                      12
                                                      %
                                    </td>
                                    <td
                                                      className="arrow"
                                    >
                                                      <i
                                                                        className="fas fa-minus grey"
                                                      />
                                    </td>
                  </tr>
</tbody>,
                <tbody>
                  <tr
                                    className="stat-line"
                  >
                                    <td
                                                      className="year"
                                    >
                                                      2005
                                    </td>
                                    <td
                                                      className="value red"
                                    >
                                                      23
                                                      %
                                    </td>
                                    <td
                                                      className="arrow"
                                    >
                                                      <i
                                                                        className="fas fa-arrow-up green"
                                                      />
                                    </td>
                  </tr>
</tbody>,
              ],
              "className": "stats-table",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": "val-2004",
                "nodeType": "host",
                "props": Object {
                  "children": <tr
                    className="stat-line"
>
                    <td
                                        className="year"
                    >
                                        2004
                    </td>
                    <td
                                        className="value red"
                    >
                                        12
                                        %
                    </td>
                    <td
                                        className="arrow"
                    >
                                        <i
                                                            className="fas fa-minus grey"
                                        />
                    </td>
</tr>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <td
                        className="year"
>
                        2004
</td>,
                      <td
                        className="value red"
>
                        12
                        %
</td>,
                      <td
                        className="arrow"
>
                        <i
                                                className="fas fa-minus grey"
                        />
</td>,
                    ],
                    "className": "stat-line",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "2004",
                        "className": "year",
                      },
                      "ref": null,
                      "rendered": "2004",
                      "type": "td",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": Array [
                          "12",
                          "%",
                        ],
                        "className": "value red",
                      },
                      "ref": null,
                      "rendered": Array [
                        "12",
                        "%",
                      ],
                      "type": "td",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": <i
                          className="fas fa-minus grey"
/>,
                        "className": "arrow",
                      },
                      "ref": null,
                      "rendered": Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "className": "fas fa-minus grey",
                        },
                        "ref": null,
                        "rendered": null,
                        "type": "i",
                      },
                      "type": "td",
                    },
                  ],
                  "type": "tr",
                },
                "type": "tbody",
              },
              Object {
                "instance": null,
                "key": "val-2005",
                "nodeType": "host",
                "props": Object {
                  "children": <tr
                    className="stat-line"
>
                    <td
                                        className="year"
                    >
                                        2005
                    </td>
                    <td
                                        className="value red"
                    >
                                        23
                                        %
                    </td>
                    <td
                                        className="arrow"
                    >
                                        <i
                                                            className="fas fa-arrow-up green"
                                        />
                    </td>
</tr>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <td
                        className="year"
>
                        2005
</td>,
                      <td
                        className="value red"
>
                        23
                        %
</td>,
                      <td
                        className="arrow"
>
                        <i
                                                className="fas fa-arrow-up green"
                        />
</td>,
                    ],
                    "className": "stat-line",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "2005",
                        "className": "year",
                      },
                      "ref": null,
                      "rendered": "2005",
                      "type": "td",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": Array [
                          "23",
                          "%",
                        ],
                        "className": "value red",
                      },
                      "ref": null,
                      "rendered": Array [
                        "23",
                        "%",
                      ],
                      "type": "td",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": <i
                          className="fas fa-arrow-up green"
/>,
                        "className": "arrow",
                      },
                      "ref": null,
                      "rendered": Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "className": "fas fa-arrow-up green",
                        },
                        "ref": null,
                        "rendered": null,
                        "type": "i",
                      },
                      "type": "td",
                    },
                  ],
                  "type": "tr",
                },
                "type": "tbody",
              },
            ],
            "type": "table",
          },
        ],
        "type": "div",
      },
    ],
    "type": "article",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="left-triangle"
/>,
          <div
            className="right-triangle"
/>,
          <div
            className="content-container"
>
            <h1
                        className="card-header normal-header"
            >
                        ASD 20
            </h1>
            <table
                        className="stats-table"
            >
                        <tbody>
                                    <tr
                                                className="stat-line"
                                    >
                                                <td
                                                            className="year"
                                                >
                                                            2004
                                                </td>
                                                <td
                                                            className="value red"
                                                >
                                                            12
                                                            %
                                                </td>
                                                <td
                                                            className="arrow"
                                                >
                                                            <i
                                                                        className="fas fa-minus grey"
                                                            />
                                                </td>
                                    </tr>
                        </tbody>
                        <tbody>
                                    <tr
                                                className="stat-line"
                                    >
                                                <td
                                                            className="year"
                                                >
                                                            2005
                                                </td>
                                                <td
                                                            className="value red"
                                                >
                                                            23
                                                            %
                                                </td>
                                                <td
                                                            className="arrow"
                                                >
                                                            <i
                                                                        className="fas fa-arrow-up green"
                                                            />
                                                </td>
                                    </tr>
                        </tbody>
            </table>
</div>,
        ],
        "className": "district-card",
        "id": "val-1",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "left-triangle",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "right-triangle",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h1
                className="card-header normal-header"
>
                ASD 20
</h1>,
              <table
                className="stats-table"
>
                <tbody>
                                <tr
                                                className="stat-line"
                                >
                                                <td
                                                                className="year"
                                                >
                                                                2004
                                                </td>
                                                <td
                                                                className="value red"
                                                >
                                                                12
                                                                %
                                                </td>
                                                <td
                                                                className="arrow"
                                                >
                                                                <i
                                                                                className="fas fa-minus grey"
                                                                />
                                                </td>
                                </tr>
                </tbody>
                <tbody>
                                <tr
                                                className="stat-line"
                                >
                                                <td
                                                                className="year"
                                                >
                                                                2005
                                                </td>
                                                <td
                                                                className="value red"
                                                >
                                                                23
                                                                %
                                                </td>
                                                <td
                                                                className="arrow"
                                                >
                                                                <i
                                                                                className="fas fa-arrow-up green"
                                                                />
                                                </td>
                                </tr>
                </tbody>
</table>,
            ],
            "className": "content-container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "ASD 20",
                "className": "card-header normal-header",
              },
              "ref": null,
              "rendered": "ASD 20",
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <tbody>
                    <tr
                                        className="stat-line"
                    >
                                        <td
                                                            className="year"
                                        >
                                                            2004
                                        </td>
                                        <td
                                                            className="value red"
                                        >
                                                            12
                                                            %
                                        </td>
                                        <td
                                                            className="arrow"
                                        >
                                                            <i
                                                                                className="fas fa-minus grey"
                                                            />
                                        </td>
                    </tr>
</tbody>,
                  <tbody>
                    <tr
                                        className="stat-line"
                    >
                                        <td
                                                            className="year"
                                        >
                                                            2005
                                        </td>
                                        <td
                                                            className="value red"
                                        >
                                                            23
                                                            %
                                        </td>
                                        <td
                                                            className="arrow"
                                        >
                                                            <i
                                                                                className="fas fa-arrow-up green"
                                                            />
                                        </td>
                    </tr>
</tbody>,
                ],
                "className": "stats-table",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": "val-2004",
                  "nodeType": "host",
                  "props": Object {
                    "children": <tr
                      className="stat-line"
>
                      <td
                                            className="year"
                      >
                                            2004
                      </td>
                      <td
                                            className="value red"
                      >
                                            12
                                            %
                      </td>
                      <td
                                            className="arrow"
                      >
                                            <i
                                                                  className="fas fa-minus grey"
                                            />
                      </td>
</tr>,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        <td
                          className="year"
>
                          2004
</td>,
                        <td
                          className="value red"
>
                          12
                          %
</td>,
                        <td
                          className="arrow"
>
                          <i
                                                    className="fas fa-minus grey"
                          />
</td>,
                      ],
                      "className": "stat-line",
                    },
                    "ref": null,
                    "rendered": Array [
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": "2004",
                          "className": "year",
                        },
                        "ref": null,
                        "rendered": "2004",
                        "type": "td",
                      },
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": Array [
                            "12",
                            "%",
                          ],
                          "className": "value red",
                        },
                        "ref": null,
                        "rendered": Array [
                          "12",
                          "%",
                        ],
                        "type": "td",
                      },
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": <i
                            className="fas fa-minus grey"
/>,
                          "className": "arrow",
                        },
                        "ref": null,
                        "rendered": Object {
                          "instance": null,
                          "key": undefined,
                          "nodeType": "host",
                          "props": Object {
                            "className": "fas fa-minus grey",
                          },
                          "ref": null,
                          "rendered": null,
                          "type": "i",
                        },
                        "type": "td",
                      },
                    ],
                    "type": "tr",
                  },
                  "type": "tbody",
                },
                Object {
                  "instance": null,
                  "key": "val-2005",
                  "nodeType": "host",
                  "props": Object {
                    "children": <tr
                      className="stat-line"
>
                      <td
                                            className="year"
                      >
                                            2005
                      </td>
                      <td
                                            className="value red"
                      >
                                            23
                                            %
                      </td>
                      <td
                                            className="arrow"
                      >
                                            <i
                                                                  className="fas fa-arrow-up green"
                                            />
                      </td>
</tr>,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        <td
                          className="year"
>
                          2005
</td>,
                        <td
                          className="value red"
>
                          23
                          %
</td>,
                        <td
                          className="arrow"
>
                          <i
                                                    className="fas fa-arrow-up green"
                          />
</td>,
                      ],
                      "className": "stat-line",
                    },
                    "ref": null,
                    "rendered": Array [
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": "2005",
                          "className": "year",
                        },
                        "ref": null,
                        "rendered": "2005",
                        "type": "td",
                      },
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": Array [
                            "23",
                            "%",
                          ],
                          "className": "value red",
                        },
                        "ref": null,
                        "rendered": Array [
                          "23",
                          "%",
                        ],
                        "type": "td",
                      },
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": <i
                            className="fas fa-arrow-up green"
/>,
                          "className": "arrow",
                        },
                        "ref": null,
                        "rendered": Object {
                          "instance": null,
                          "key": undefined,
                          "nodeType": "host",
                          "props": Object {
                            "className": "fas fa-arrow-up green",
                          },
                          "ref": null,
                          "rendered": null,
                          "type": "i",
                        },
                        "type": "td",
                      },
                    ],
                    "type": "tr",
                  },
                  "type": "tbody",
                },
              ],
              "type": "table",
            },
          ],
          "type": "div",
        },
      ],
      "type": "article",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
